airbyte-integrations/connectors/source-posthog/source_posthog/streams.py

```diff
@@ -0,0 +1,306 @@
+                return params
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        """
+        Return the latest state by comparing the cursor value in the latest record with the stream's most recent state object
+        and returning an updated state object.
+        """
+        current_stream_state = current_stream_state or {}
+        latest_state = latest_record.get(self.cursor_field)
+        current_state = current_stream_state.get(self.cursor_field) or latest_state
+        return {self.cursor_field: max(latest_state, current_state)}
+
+    def parse_response(self, response: requests.Response, stream_state: Mapping[str, Any], **kwargs) -> Iterable[Mapping]:
+        response_json = response.json()
+        data = response_json.get(self.data_field)
+        if data and stream_state:
+            last_record_curfield = data[-1][self.cursor_field]
+            last_stream_curfield = stream_state[self.cursor_field]
+            if last_stream_curfield >= last_record_curfield:
+                yield from []
+            else:
+                first_record_curfield = data[0][self.cursor_field]
+                if first_record_curfield > last_stream_curfield:
+                    yield from data
+                else:
+                    slice_ = [i for i in data if i[self.cursor_field] > last_stream_curfield]
+                    yield from slice_
+
+        else:
+            yield from data
+
+
+class Elements(PosthogStream):
+    data_field = "elements"
+
+    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
+        return "element/stats"
+
+    def parse_response(self, response: requests.Response, stream_state: Mapping[str, Any], **kwargs) -> Iterable[Mapping]:
+        response_json = response.json()
+        elements = [i[self.data_field][0] for i in response_json]
+        yield from elements
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        """
+        This enpoint returns full stats list and have no pagination.
+        """
+        return None
+
+
+class Events(IncrementalPosthogStream):
+    """
+    CAUTION! API supports inserting custom timestamp.
+    That may break incremental since there is no way to sort in request params.
+    """
+
+    cursor_field = "timestamp"
+
+    def __init__(self, start_date: str, **kwargs):
+        self.start_date = start_date
+        super().__init__(**kwargs)
+
+    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
+        return "event"
+
+    def request_params(self, stream_state=None, stream_slice: Mapping[str, Any] = None, **kwargs):
+        stream_state = stream_state or {}
+        params = super().request_params(stream_state=stream_state, **kwargs)
+        since_value = stream_state.get(self.cursor_field) or self.start_date
+        since_value = max(since_value, self.start_date)
```

Comment:
Are you comparing strings?
This does not always lead to the correct result. It may be better to compare the datetime data types from Pendulum

Reply:
ISO format, get it to keep as string